### DISCUSSION TITLE
Separate usage of the Lighthouse positioning deck.

### DISCUSSION THREAD

**Discussion by ivomarvan** [bitcraze/discussions#291] [Q&A] [answered]
I bought the Lighthouse positioning deck without other parts from the Crazyflie line.

1) Can I expect some reasonable output on the UART port even without configuration? (i.e. DEV RX, DEV TX, UART pads available to solder and 2.54mm pin header for connection to an external system).
Is the protocol on this port described somewhere?

2) Can I configure (with your software) using a platform other than Crazyflie? (Perhaps RPI, ESP32, etc.?)

Thank you, and have a nice day.

**Comment by krichardsson** [comment, marked as answer]
1. No
2. Yes, it is possible but a bit complex. I'm not that good at the details but I can outline what needs to be done

The boot sequence is implemented in `lighthouseDeckFlasherCheckVersionAndBoot()`, [lighthouse_deck_flasher.c](https://github.com/bitcraze/crazyflie-firmware/blob/master/src/modules/src/lighthouse/lighthouse_deck_flasher.c)

## Flash a new binary to the FPGA

The binary that is on the deck when shipped is old and I don't think it will do the job (it is mainly used to verify the decks in production).
In the boot sequence this will give the wrong check sum for the binary and leave the deck in bootloader mode to enable flashing of a new binary. What you need to do is essentially to
1. Boot the deck and set it in bootloader mode. This is done via I2C
2. Flash the latest released binary from https://github.com/bitcraze/lighthouse-fpga

## Boot the deck
 
1. Start it in "normal" mode (see 
2. You will get sweep data on the UART

The sweep data is unfortunately not delivered in a very simple format and there is a brief documentation in https://github.com/bitcraze/lighthouse-fpga. You can also find some tools and related code in this repo, I think most of it is in a fairly working state.
It is not obvious how the UART protocol is working (there are some corner cases) but you can always look at the code in the Crazyflie that is decoding it. [LH1 decoder](https://github.com/bitcraze/crazyflie-firmware/blob/master/src/utils/src/lighthouse/pulse_processor_v1.c), [LH2 decoder](https://github.com/bitcraze/crazyflie-firmware/blob/master/src/utils/src/lighthouse/pulse_processor_v2.c)

**Reply by ivomarvan** [reply]
Hi Kristoffer ([krichardsson](https://github.com/krichardsson)),
Thank you for your quick reply and useful information!

It looks like a lot of work, and a big challenge :-).

I note that it is a bit confusing to state "UART pads available to solder a 2.54mm pin header for connection to an external system, eg. Arduino" (e.g. in https://www.bitcraze.io/products/lighthouse-positioning-deck ) when it can't really be used much.

Have a nice day.
Ivo

**Reply by tobbeanton** [reply]
We have been discussing extending the usage of the lighthouse deck to be more stand alone but it is challenging. Many tools and algorithms that need to run on different platforms...

I agree that the "UART pads..." can be a bit confusing but it is under the Electrical specification and only descries the electrical properties. Clarifying it that there is no software support wouldn't harm though. We will add that to the TODO list.

**Reply by ivomarvan** [reply]
If I could choose such a "more stand-alone device", I would just want a sequence of numbers on the UART (or any)  port that would represent the flash length and inter-flash interval for individual TS4231 devices. Plus independent configurable software (e.g. in python) that would calculate the position with any external hardware.

For someone (like you), this hardware can be implemented as an FPGA, for someone, for example, in a Jetson NVIDIA computer, and for someone, non-real-time processing on a desktop would be enough.

**Reply by krichardsson** [reply]
Thanks for the feedback!

**Comment by MarcosJNS** [comment]
Hi @ivomarvan @krichardsson @tobbeanton my team and I are facing a similar issue when trying to follow the documentation to use the Lighthouse positioning deck as stand alone positioning system alongside with a raspberry pi. Has any progress been made on this regard?
Any information regarding to the integration process is very welcome.
Thanks

**Reply by ivomarvan** [reply]
Hi @MarcosJNS 
sorry for the late reply, I was on vacation.
We have put this matter on hold at the company for some time, but we are revisiting it in our spare time and I have some new insights.
When I find some time, I will describe it here.
